frontend/src/context/analysisResultStateSlice.ts

```diff
@@ -737,7 +737,8 @@ export const requestAndStoreAnalysis = createAsyncThunk<
   );
 
   // Create a legend based on statistic data to be used for admin level analsysis.
-  const legend = createLegendFromFeatureArray(features, statistic);
+  const legend =
+    hazardLayer.legend ?? createLegendFromFeatureArray(features, statistic);
```

Comment:
In theory this would also depend on the type of `statistic` being run. This makes sense for `min, max, mean, median` but not so much for `sum, std, count`

Maybe add a small note for later? Or a switch on statistics? wdyt?